### DISCUSSION TITLE
Running in Production with PM2?

### DISCUSSION THREAD

**Discussion by osseonews** [vendurehq/vendure#947] [General]
This is easy to run standalone and develop on. However, to run a production website, you need to have a Process Manager so the app continues to run. The popular process manager for Node is PM2 (https://pm2.keymetrics.io/). What command do we use to run this with PM2? PM2 expects a script as an argument, but there is no script in the main folder. We've tried `pm2 start ./src/index.ts`  and other variations, but have no success. Anyone now how to run this with PM2 or any other process manager for that matter? Without instructions for this, you can't actually run a production server with this app, which is the goal, of course. Thanks.

**Comment by michaelbromley** [comment]
Hi, I agree that something like pm2 makes sense for production. Another option is to use Docker which can also cover many of the benefits of pm2. Also you can use both together: https://pm2.keymetrics.io/docs/usage/quick-start/

Anyway, the issue at your end seems to be that you are attempting to run a TypeScript file with pm2. This will not work because pm2 is expecting a JavaScript file. So you'll need to compile your source files to JavaScript using the TS compiler (`npm run tsc`) and then run pm2 on the resulting `./dist/index.js` file (exact location depends on your tsconfig).

**Reply by osseonews** [reply]
Thanks! I was thinking it was a TS issue. Will try your fix and let you know.

**Reply by osseonews** [reply]
Couldn't get it to work with TS. So we just installed the JS version of Vendure and PM2 works fine. Thanks.
